public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -350,32 +350,17 @@ export class GraphiteDatasource extends DataSourceApi<GraphiteQuery, GraphiteOpt
     );
 
     // special handling for tag_values(<tag>[,<expression>]*), this is used for template variables
-    let matches = interpolatedQuery.match(/^tag_values\(([^,]+)((, *[^,]+)*)\)$/);
-    if (matches) {
-      const expressions = [];
-      const exprRegex = /, *([^,]+)/g;
-      let match = exprRegex.exec(matches[2]);
-      while (match !== null) {
-        expressions.push(match[1]);
-        match = exprRegex.exec(matches[2]);
-      }
+    let allParams = interpolatedQuery.match(/^tag_values\((.*)\)$/);
+    let expressions = allParams ? allParams[1].split(',').filter((p) => !!p) : undefined;
```

Comment:
Filtering is used to handle cases where no params are passed, e.g. `tag_values()`